sensors/test_deps.go

```diff
@@ -27,47 +28,103 @@ var (
+	GoodLidar TestSensor = "good_lidar"
+	// WarmingUpLidar is a lidar whose NextPointCloud function returns a "warming up" error.
+	WarmingUpLidar TestSensor = "warming_up_lidar"
+	// LidarWithErroringFunctions is a lidar whose functions return errors.
+	LidarWithErroringFunctions TestSensor = "lidar_with_erroring_functions"
+	// LidarWithInvalidProperties is a lidar whose properties are invalid.
+	LidarWithInvalidProperties TestSensor = "lidar_with_invalid_properties"
+	// GibberishLidar is a lidar that can't be found in the dependencies.
+	GibberishLidar TestSensor = "gibberish_lidar"
+	// NoLidar is a lidar that represents that no lidar is set up or added.
+	NoLidar TestSensor = ""
+
+	// ReplayLidar is a lidar that works as expected and returns a pointcloud.
+	ReplayLidar TestSensor = "replay_lidar"
+	// InvalidReplayLidar is a lidar whose meta timestamp is invalid.
+	InvalidReplayLidar TestSensor = "invalid_replay_lidar"
+	// FinishedReplayLidar is a lidar whose NextPointCloud function returns an end of dataset error.
+	FinishedReplayLidar TestSensor = "finished_replay_lidar"
+
+	// GoodIMU is an IMU that works as expected and returns linear acceleration and angular velocity values.
+	GoodIMU TestSensor = "good_imu"
+	// IMUWithErroringFunctions is an IMU whose functions return errors.
+	IMUWithErroringFunctions TestSensor = "imu_with_erroring_functions"
+	// IMUWithInvalidProperties is an IMU whose properties are invalid.
+	IMUWithInvalidProperties TestSensor = "imu_with_invalid_properties"
+	// NoIMU is an IMU that represents that no IMU is set up or added.
+	NoIMU TestSensor = ""
+
+	// ReplayIMU is an IMU that works as expected and returns linear acceleration and angular velocity values.
+	ReplayIMU TestSensor = "replay_imu"
+	// InvalidReplayIMU is an IMU whose meta timestamp is invalid.
+	InvalidReplayIMU TestSensor = "invalid_replay_imu"
+	// FinishedReplayIMU is an IMU whose LinearAcceleration and AngularVelocity functions return an end of
+	// dataset error.
+	FinishedReplayIMU TestSensor = "finished_replay_imu"
+
+	// GoodOdometer is an odometer that works as expected and returns position and orientation values.
+	GoodOdometer TestSensor = "good_odometer"
+
+	// MovementSensorNotIMUNotOdometer is a movement sensor that does neither support an IMU nor an odometer.
+	MovementSensorNotIMUNotOdometer TestSensor = "movement_sensor_not_imu_not_odometer"
+
+	// MovementSensorBothIMUAndOdometer is a movement sensor that dsupports both an IMU nor an odometer.
+	MovementSensorBothIMUAndOdometer TestSensor = "movement_sensor_imu_and_odometer"
+	// GibberishMovementSensor is a movement sensor that can't be found in the dependencies.
+	GibberishMovementSensor TestSensor = "gibberish_movement_sensor"
+	// MovementSensorWithErroringPropertiesFunc is a movement sensor whose Properties function returns an error.
+	MovementSensorWithErroringPropertiesFunc TestSensor = "movement_sensor_with_erroring_properties_function"
+)
```

Comment:
Refactoring: Using variables rather than strings to identify mock sensors makes it easier to follow & read the code.